opentracing-jdbc/opentracing-jdbc/src/main/java/org/zalando/opentracing/jdbc/span/DatabaseInstanceSpanDecorator.java

```diff
@@ -0,0 +1,47 @@
+package org.zalando.opentracing.jdbc.span;
+
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+import org.apiguardian.api.API;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+
+import static org.apiguardian.api.API.Status.EXPERIMENTAL;
+
+/**
+ * Sets the <code>db.instance</code> span tag.
+ *
+ * @see <a href="https://opentracing.io/specification/conventions/#span-tags-table">Standard Span Tags</a>
+ */
+@API(status = EXPERIMENTAL)
+public final class DatabaseInstanceSpanDecorator implements SpanDecorator {
+
+    @Override
+    public void onQuery(final Span span, final Statement statement, final List<String> queries) throws SQLException {
+        final String url = statement.getConnection().getMetaData().getURL();
+
+        /*
+         * TODO This is not ideal, since according to the spec we should be
```

Comment:
any risk to accidentally expose credentials here?

Reply:
Seems like you're right. JDBC doesn't specify this and some drivers do support it. What's the best course of action now?

1. Get rid of this decorator completely?
2. Attempt to parse the database name out of the URL?
3. Keep the decorator but disable it by default?

Reply:
2nd seems reasonable to me

Reply:
Yep, I'll change it.

Reply:
Turns out that [`Connection.getCatalog()`](https://docs.oracle.com/javase/8/docs/api/java/sql/Connection.html#getCatalog--) is what I was looking for. :man_dancing: